include/deal.II/meshworker/mesh_loop.h

```diff
@@ -0,0 +1,232 @@
+
+
+#ifndef dealii__mesh_loop_h
+#define dealii__mesh_loop_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/work_stream.h>
+#include <deal.II/base/template_constraints.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/filtered_iterator.h>
+#include <deal.II/meshworker/loop.h>
+#include <deal.II/meshworker/local_integrator.h>
+#include <deal.II/meshworker/dof_info.h>
+#include <deal.II/meshworker/integration_info.h>
+#include <deal.II/meshworker/assemble_flags.h>
+
+#include <functional>
+
+DEAL_II_NAMESPACE_OPEN
+
+template <typename> class TriaActiveIterator;
+
+namespace MeshWorker
+{
+  /**
+   * A low level work function of this namespace.
+   *
+   * @ingroup MeshWorker
+   * @author Luca Heltai, 2017
+   */
+  template <class Iterator,
+            class CellWorker, class Copyer,
+            class ScratchData, class CopyData,
+            class BoundaryWorker, class FaceWorker>
+  void mesh_loop(const Iterator & begin,
+                 const typename identity<Iterator>::type &end,
+
+//                 const std::function<void (const Iterator &, ScratchData &, CopyData &)> &cell_worker,
+//                 const std::function<void (const CopyData &)> &copyer,
+                 CellWorker cell_worker,
+                 Copyer copyer,
+
+                 ScratchData &scratch_data,
+                 CopyData &copy_data,
+
+                 const AssembleFlags flags = assemble_own_cells,
+
+//                 const std::function<void (const Iterator &, const unsigned int&, ScratchData &, CopyData &)> &boundary_worker=
+//                  std::function<void (const Iterator &, const unsigned int&, ScratchData &, CopyData &)>(),
+
+//                 const std::function<void (const Iterator &, const unsigned int &, const unsigned int &,
+//                                           const Iterator &, const unsigned int &, const unsigned int &,
+//                                           ScratchData &, CopyData &)> &face_worker=
+//                  std::function<void (const Iterator &, unsigned int, unsigned int,
+//                                      const Iterator &, unsigned int, unsigned int,
+//                                      ScratchData &, CopyData &)>(),
+
+                 BoundaryWorker boundary_worker = BoundaryWorker(),
+                 FaceWorker face_worker = FaceWorker(),
+
+                 const unsigned int 	queue_length = 2*MultithreadInfo::n_threads(),
+                 const unsigned int 	chunk_size = 8)
+  {
+    auto cell_action = [&] (const Iterator &cell, ScratchData &scratch, CopyData &copy)
+    {
+      const bool ignore_subdomain = (cell->get_triangulation().locally_owned_subdomain()
+                                     == numbers::invalid_subdomain_id);
+
+      types::subdomain_id csid = /*(cell->is_level_cell())
+                                 ? cell->level_subdomain_id()
+                                 : */cell->subdomain_id();
```

Comment:
can you make the variable `const`? I'm not fond of the name -- maybe `current_subdomain_id`?